foundryup/foundryup

```diff
@@ -23,6 +24,8 @@ main() {
       -r|--repo)        shift; FOUNDRYUP_REPO=$1;;
       -b|--branch)      shift; FOUNDRYUP_BRANCH=$1;;
       -v|--version)     shift; FOUNDRYUP_VERSION=$1;;
+      -l|--list)        shift; list;;
```

Comment:
If you attempt to install an invalid version, like:

```
foundryup --version 59f354c179f4e7f6d7292acb3d068815c79286d1
```

or 

```
foundryup --version https://github.com/foundry-rs/foundry/archive/refs/tags/nightly-aa69ed1e46dd61fbf9d73399396a4db4dd527431.tar.gz
```

it won't exit cleanly meaning when you run `--list` it returns

```
foundryup: https:
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/https:/forge: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/https:/forge --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/https:/cast: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/https:/cast --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/https:/anvil: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/https:/anvil --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/https:/chisel: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/https:/chisel --version
foundryup: - 
foundryup: ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/forge: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/forge --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/cast: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/cast --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/anvil: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/anvil --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/chisel: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b/chisel --version
foundryup: - 
foundryup: nightly
foundryup: - forge 0.2.0 (2f698e4 2024-12-13T06:35:25.027879296Z)
foundryup: - cast 0.2.0 (2f698e4 2024-12-13T06:35:25.025215009Z)
foundryup: - anvil 0.2.0 (2f698e4 2024-12-13T06:35:25.023959451Z)
foundryup: - chisel 0.2.0 (2f698e4 2024-12-13T06:35:25.026724926Z)
foundryup: nightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b
foundryup: - forge 0.2.0 (e5dbb7a 2024-12-03T00:24:42.577356416Z)
foundryup: - cast 0.2.0 (e5dbb7a 2024-12-03T00:24:42.537035487Z)
foundryup: - anvil 0.2.0 (e5dbb7a 2024-12-03T00:24:42.571016376Z)
foundryup: - chisel 0.2.0 (e5dbb7a 2024-12-03T00:24:42.529565979Z)
foundryup: nightly-f089dff1c6c24d1ddf43c7cbefee46ea0197c88f
foundryup: - forge 0.2.0 (f089dff 2024-10-06T00:22:11.758232611Z)
foundryup: - cast 0.2.0 (f089dff 2024-10-06T00:22:11.750238669Z)
foundryup: - anvil 0.2.0 (f089dff 2024-10-06T00:22:11.752640285Z)
foundryup: - chisel 0.2.0 (f089dff 2024-10-06T00:22:11.701654109Z)
foundryup: v59f354c179f4e7f6d7292acb3d068815c79286d1
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/forge: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/forge --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/cast: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/cast --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/anvil: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/anvil --version
foundryup: - 
./foo.sh: line 325: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/chisel: No such file or directory
foundryup: command failed: /home/zerosnacks/.foundry/versions/v59f354c179f4e7f6d7292acb3d068815c79286d1/chisel --version
foundryup: - 
```

Reply:
```
~/.foundry/versions$ ls
https:                                           nightly                                           nightly-f089dff1c6c24d1ddf43c7cbefee46ea0197c88f
ightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b  nightly-e5dbb7a320c2b871c4a4a1006ad3c15a08fcf17b  v59f354c179f4e7f6d7292acb3d068815c79286d1
```

Reply:
fixed in https://github.com/foundry-rs/foundry/pull/9551/commits/bf24bd9155a42c152d290771f0a0496ada8ee46a

Reply:
Nice, confirming this has been fixed